src/basic/efivars.c

```diff
@@ -56,17 +61,25 @@ int efi_get_variable(
                 return -ENOMEM;
 
         if (!ret_value && !ret_size && !ret_attribute) {
-                /* If caller is not interested in anything, just check if the variable exists and is readable
-                 * to us. */
+                /* If caller is not interested in anything, just check if the variable exists and is
+                 * readable. */
                 if (access(p, R_OK) < 0)
                         return -errno;
 
                 return 0;
         }
 
-        fd = open(p, O_RDONLY|O_NOCTTY|O_CLOEXEC);
-        if (fd < 0)
-                return -errno;
+        for (unsigned try = 0;; try++) {
+                fd = open(p, O_RDONLY|O_NOCTTY|O_CLOEXEC);
```

Comment:
doing the loop around open() is unnecessary apparently, see the kernel patch that is causing this, which only added this to read()